src/filters/dasher.c

```diff
@@ -1869,6 +1871,9 @@ static Bool dasher_same_adaptation_set(GF_DasherCtx *ctx, GF_DashStream *ds, GF_
 	//not the same roles
 	if (!dasher_same_roles(ds, ds_test)) return GF_FALSE;
 
+	//intra-only trick mode belongs to a separate AS
+	if (ds->hls_intra_only != ds_test->stream_type) return GF_FALSE;
```

Comment:
shouldn't that be 
` if (ds->hls_intra_only != ds_test->hls_intra_only) return GF_FALSE;`